src/qibo/models/circuit.py

```diff
@@ -1174,7 +1177,10 @@ def to_qasm(self):
 
             qubits = ",".join(f"q[{i}]" for i in gate.qubits)
             if isinstance(gate, gates.ParametrizedGate):
-                params = (str(x) for x in gate.parameters)
+                if any(x.__class__.__name__ == "Tensor" for x in gate.parameters):
+                    params = (str(x.detach().item()) for x in gate.parameters)
+                else:
+                    params = (str(x) for x in gate.parameters)
```

Comment:
Simply `str(float(x))` was causing problems with any backend? Because, for `torch` it seems to be fine:
``` python
x = torch.tensor([2.], requires_grad=True)
y = 2 * x **2
str(float(y)) # --> "8.0"
```

Reply:
This is raising an error in test_models_circuit_qasm_cirq.py

Reply:
I cannot reproduce the error
``` python
params = (str(float(x)) for x in gate.parameters)
```
this works for me locally, not sure, is the test random? Maybe there are just some random configurations causing problems

Reply:
I get this error:
FAILED test_models_circuit_qasm_cirq.py::test_parametrized_gate_cirq[pytorch] - cirq.contrib.qasm_import.exception.QasmException: Syntax error: ','
There should not be anything random

Reply:
weird, it does not fail for me, I just checked again. Does it fail here in the CI or locally for you?
@renatomello does it fail for you as well?

Reply:
it fails locally, the parameter becomes: "torch.Tensor([....], gradient....)"

Reply:
But it also fails here, look at the commits before restoring it.
![Screenshot from 2024-10-04 12-38-31](https://github.com/user-attachments/assets/486ab6f3-b21f-4560-aeff-e8107f8a03af)